pallets/identity/src/lib.rs

```diff
@@ -1695,11 +1708,43 @@ impl<T: Trait> Module<T> {
         expiry: Option<T::Moment>,
     ) -> DispatchResult {
         Self::ensure_authorized_cdd_provider(issuer)?;
+        // Ensure cdd_id uniqueness for a given target DID.
+        Self::ensure_cdd_id_uniqueness(&claim, target)?;
 
         Self::base_add_claim(target, claim, issuer, expiry);
         Ok(())
     }
 
+    /// Enforce CDD_ID uniqueness for a given target DID.
+    ///
+    /// # Errors
+    /// - `CDDIdNotUniqueForIdentity` is returned when new cdd claim's cdd_id
+    /// doesn't matches with the existing cdd claim's cdd_id.
+    fn ensure_cdd_id_uniqueness(claim: &Claim, target: IdentityId) -> DispatchResult {
+        if let Claim::CustomerDueDiligence(cdd_id) = claim {
+            let systematic_ids = SystematicIssuers::into_enum_iter()
+                .map(|si| si.as_id())
+                .collect::<Vec<IdentityId>>();
+            ensure!(
+                !Self::base_fetch_valid_cdd_claims(target, 0.into(), None)
+                    .iter()
+                    .any(|id_claim| {
+                        if let Claim::CustomerDueDiligence(c_id) = id_claim.claim {
+                            // Allowing default cdd_id.
+                            if !c_id.is_default_cdd()
+                                && !systematic_ids.contains(&id_claim.claim_issuer)
```

Comment:
```suggestion
                                && !SYSTEMATIC_ISSUERS.iter().any(|si| si.as_id() == id_claim.claim_issuer)
```

Reply:
ahh, I missed it that we have this array already.